agent-packages/packages/salesforce/src/opportunity-tools.ts

```diff
@@ -22,48 +21,70 @@ const opportunitySearchSchema = z.object({
     .describe('The ID of the opportunity owner to filter by')
 });
 
+export const addNoteToOpportunitySchema = z.object({
+  opportunityId: z.string().describe('The ID of the opportunity to add a note to'),
+  note: z.string().describe('The content of the note to add'),
+  title: z
+    .string()
+    .nullish()
+    .transform((val) => val ?? undefined)
+    .describe('Optional title for the note')
+});
+
+export const getOpportunityUrlSchema = z.object({
+  opportunityId: z.string().describe('The ID of the opportunity to get the URL for')
+});
+
+export const getOpportunityStagesSchema = z.object({});
+
 export const opportunityTools = (config: SalesforceConfig) => {
   const service = new SalesforceOpportunityService(config);
   return [
-    tool(async (args: SearchOpportunitiesParams) => service.searchOpportunities(args), {
+    tool({
       name: 'salesforce_search_opportunities',
       description:
         'Search for opportunities in Salesforce based on keyword, stage, and/or owner ID',
-      schema: opportunitySearchSchema
+      schema: opportunitySearchSchema,
+      operations: [ToolOperation.READ],
+      func: async (args: z.infer<typeof opportunitySearchSchema>) => {
+        return await service.searchOpportunities(args);
+      }
     }),
-    tool(
-      async (args: { opportunityId: string; note: string; title?: string }) =>
-        service.addNoteToOpportunity(args.opportunityId, args.note, args.title),
-      {
-        name: 'salesforce_add_note_to_opportunity',
-        description: 'Add a note to a Salesforce opportunity',
-        schema: z.object({
-          opportunityId: z.string().describe('The ID of the opportunity to add a note to'),
-          note: z.string().describe('The content of the note to add'),
-          title: z
-            .string()
-            .nullish()
-            .transform((val) => val ?? undefined)
-            .describe('Optional title for the note')
-        })
+    tool({
+      name: 'salesforce_add_note_to_opportunity',
+      description: 'Add a note to a Salesforce opportunity',
+      schema: addNoteToOpportunitySchema,
+      operations: [ToolOperation.CREATE],
+      func: async (args: z.infer<typeof addNoteToOpportunitySchema>) => {
+        return await service.addNoteToOpportunity(args.opportunityId, args.note, args.title);
       }
-    ),
-    tool(async (args: { opportunityId: string }) => service.getOpportunityUrl(args.opportunityId), {
+    }),
+    tool({
       name: 'salesforce_get_opportunity_url',
       description: 'Get the URL to view an opportunity in Salesforce',
-      schema: z.object({
-        opportunityId: z.string().describe('The ID of the opportunity to get the URL for')
-      })
+      schema: getOpportunityUrlSchema,
+      operations: [ToolOperation.READ],
+      func: (args: z.infer<typeof getOpportunityUrlSchema>) => {
+        return Promise.resolve(service.getOpportunityUrl(args.opportunityId));
+      }
     }),
```

Comment:
You can do this instead of using Promise.resolve
```suggestion
      func: async (args: z.infer<typeof getOpportunityUrlSchema>) => {
        return service.getOpportunityUrl(args.opportunityId);
      }
    }),
```